v1.4.0.rc1

Macr0Nerd opened the pull request:
### Added
- **Modify** - Added the `modify` command
- **Create** - Added multi-architecture support
- **Configuration** - Added multi-architecture AMI configuration

### Changed
- **Create** - Start the master and worker at the same time
- **Create** - Change how the capacity unit type is specified to better allow `modify` to work
- **Documentation** - Update documentation

Macr0Nerd committed:
Merge pull request #65 from carsdotcom/dev

v1.3.3

Macr0Nerd committed:
Bump version to 1.4.0

Macr0Nerd committed:
Implement instance_type
Fix tests for instance_type

Full instance type implementation

Basic instance type implemented

Macr0Nerd committed:
Start master and worker at the same time

Add todos

Macr0Nerd committed:
Add Forge modify

Macr0Nerd committed:
Fix tests

Macr0Nerd committed:
Fix incorrect name attribute

Macr0Nerd committed:
Update master and single capacity unit

Currently, master and single instances specify that they want a single instance with the specified resources.
This is done using the fleet capacity unit 'units'.
We can instead specify the minimum resources we want and set the minimum resources of the instance above this so that a single instance fulfills the reservation.
This means we would specify the fleet capacity unit as 'memory-mib'.
This difference is important as we cannot change the capacity unit with forge modify, so to scale up/down resources we can only change the total capacity.
For master and single instances, setting the capacity unit to 'units' means we effectively lose the ability to modify the fleet as we do not want more units, but more resources.
For workers this is less of an issue but we should prefer 'memory-mib'.
This commit sets it so that single and master instances use 'memory-mib' unless otherwise specified.

Macr0Nerd committed:
Merge pull request #69 from carsdotcom/dev

Forge v1.3.5

Macr0Nerd committed:
Merge branch 'main' into v1.4.0-dev

Macr0Nerd committed:
Add multi-architecture support

Macr0Nerd committed:
Add IMDSv1 warning to create.py

Macr0Nerd committed:
Add experimental warning to modify.py

Macr0Nerd committed:
Update documentation

Macr0Nerd committed:
Update CHANGELOG.md

Macr0Nerd committed:
Bump version to rc1

jagmoreira commented on src/forge/create.py:
```diff
@@ -576,11 +628,14 @@ def create_fleet(n, config: Configuration, task, instance_details):
     now_utc = now_utc.replace(microsecond=0)
     valid_until = now_utc + timedelta(hours=int(valid))
     subnet = config.aws_multi_az
+    instance_type = config.instance_type
```
It doesn't look like this is being used

jagmoreira commented on docs/modify.md:
```diff
@@ -0,0 +1,39 @@
+
+### Parameters
+
+#### Required 
```
It would be good to include which parameters cannot be modified like name, service, etc.

Macr0Nerd committed:
Fix bug in IMDS warning

Macr0Nerd committed:
Remove unnecessary dependency

Macr0Nerd committed:
Set minimum boto3 version

Macr0Nerd committed:
Set minimum pyyaml version

Macr0Nerd commented:
This has been converted to a draft as it is waiting on a resolution from AWS regarding multiarchitecture fleets.

Macr0Nerd committed:
Update Forge multi-architecture handling

The way that multi-architecture support was
added in v1.4.0rc1 did not adequately
construct the fleet request. After
discussing with AWS Support, the recommended
configuration has been implemented in this
update. In testing so far this has still
not worked as expected. There is a support
case open with AWS regarding this, but for
now Forge implements what is best practice
to our knowledge.

Macr0Nerd committed:
Add cloudtrail error checking to create

Macr0Nerd committed:
Update cloudtrail filter names

Macr0Nerd committed:
Bump version 1.4.0rc3

Macr0Nerd committed:
Add AMI search filters

Macr0Nerd committed:
Add role mangling toggle

This option is predominantly for Forge deployments where the Forge roles do not follow the contemporary Forge role naming convention. Current versions of Forge wrap the role name as 'forge-aws_role-env', but this is not always desired. The wrap_role option defaults to True as to not change the existing behaviour, but now allows for this to be disabled.

Macr0Nerd committed:
Update documentation

Macr0Nerd committed:
Bump version to 1.4.1

Macr0Nerd committed:
Merge branch 'dev' into v1.4.1

# Conflicts:
#	.bumpversion.cfg
#	CHANGELOG.md
#	src/forge/__init__.py

Macr0Nerd committed:
v1.4.0.dev4

Macr0Nerd committed:
Add more nuanced AMI tag filtering

Macr0Nerd committed:
Bump to v1.4.0.dev5

Macr0Nerd committed:
Fix lowercase log level bug

Macr0Nerd committed:
v1.4.1

Macr0Nerd committed:
Fix pricing bug
